sql/core/src/test/scala/org/apache/spark/sql/execution/OptimizeMetadataOnlyQuerySuite.scala

```diff
@@ -150,4 +150,30 @@ class OptimizeMetadataOnlyQuerySuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-31590 The filter used by Metadata-only queries should not have Unevaluable") {
+    withTable("test_tbl") {
+      withSQLConf(OPTIMIZER_METADATA_ONLY.key -> "true") {
+        sql("CREATE TABLE test_tbl (a INT,d STRING,h STRING) USING PARQUET PARTITIONED BY (d ,h)")
```

Comment:
Can you make the test case minimised, and consistent with the style used in this file? I think you can create the partitioned table via `write.parquetby` syntax instead of relying on the SQL syntax here.